pkg/middleware/basic_auth.go

```diff
@@ -17,40 +17,61 @@
 package middleware
 
 import (
-	"github.com/Peripli/service-manager/pkg/log"
-	"net/http"
-
-	"github.com/Peripli/service-manager/pkg/util"
+	"context"
+	"fmt"
+	"github.com/Peripli/service-manager/api/filters/authn"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
 )
 
-const (
-	notAuthorized = "Not Authorized"
-	errorMessage  = "Unauthorized resource access"
-)
+type basicAuthFilter struct {
+	web.Filter
+}
 
-// BasicAuth is a middleware for basic authorization
-func BasicAuth(username, password string) func(handler http.Handler) http.Handler {
-	return func(handler http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			if !authorized(r, username, password) {
-				logger := log.C(r.Context())
-				logger.WithField("username", username).Debug(errorMessage)
-				err := util.WriteJSON(w, http.StatusUnauthorized, &util.HTTPError{
-					ErrorType:   notAuthorized,
-					Description: errorMessage,
-					StatusCode:  http.StatusUnauthorized,
-				})
-				if err != nil {
-					logger.Error(err)
-				}
-				return
-			}
-			handler.ServeHTTP(w, r)
-		})
+func BasicFilter(username, password string) web.Filter {
```

Comment:
[warning]: pkg/middleware/basic_auth.go: 32:1: exported function BasicFilter should have comment or be unexported (golint)